pyogrio/_io.pyx

```diff
@@ -376,6 +378,33 @@ cdef get_feature_count(OGRLayerH ogr_layer):
     return feature_count
 
 
+cdef get_total_bounds(OGRLayerH ogr_layer, int force):
+    """Get the total bounds of a layer.
+
+    Parameters
+    ----------
+    ogr_layer : pointer to open OGR layer
+    force : bool
+        True if the total bounds should be computed even if it is expensive
+
+    Returns
+    -------
+    tuple of (xmin, ymin, xmax, ymax) or None
+        The total bounds of the layer, or None if they could not be determined.
+    """
+
+    cdef OGREnvelope ogr_envelope # = NULL
+    try:
+        exc_wrap_ogrerr(OGR_L_GetExtent(ogr_layer, &ogr_envelope, force))
+        bounds = (
```

Comment:
I imagine there could be a slight though unlikely chance that `ogr_envelope` remains `NULL` without an error being raised, so I'd suggest a check for that before extracting bounds:

```
if ogr_envelope == NULL:
    bounds = None
else:
    bounds = ...

```

Reply:
ok